### ISSUE TITLE
Unable to access frontend due to proxy server error on new deployment

### ISSUE THREAD

**Issue by er42** [onnela-lab/beiwe-backend#302]
I'm trying to deploy a new Beiwe cluster. I went through all the steps in the current deployment instructions, and everything appeared to work fine. However, there seems to be something wrong with the Apache server configuration and I'm unable to reach the frontend site. Instead, I'm getting the following 502 Proxy error:

_**Proxy Error**
The proxy server received an invalid response from an upstream server.
The proxy server could not handle the request
Reason: Error reading from remote server_

I'm not that familiar with Apache, but I've been trying to do some basic troubleshooting...

My attempts to access the site do seem to be showing up on the server.  When I go to our URL (which is pointing to the load balancer via Route 53 A and AAAA records), my attempt to reach the server does show up in `/var/log/httpd/access_log` on the EC2 instance, with a record like:
`"GET /favicon.ico HTTP/1.1" 502 - "https://studies.OURDOMAIN.net/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/106.0.0.0 Safari/537.36"`

However, `/var/log/httpd/error_log` is full of errors like:
`[Thu Nov 03 21:12:32.113460 2022] [proxy:error] AH00898: Error reading from remote server returned by /
[Thu Nov 03 21:12:32.174226 2022] [proxy_http:error] (20014)Internal error (specific information not available): AH01102: error reading status line from remote server localhost:8000, referer: https://studies.OURDOMAIN.net/`

I've double-checked that there is an open HTTPS listener on the load balancer, that the relevant security groups are set to allow incoming HTTPS, and that the SSL certificates are valid and attached.

Weirdly, when I do `curl 127.0.0.1` on the EC2 server itself, I get the exact same Error 502. So, I'm thinking this is probably something to do with the internal configuration of the server rather than a closed port issue, right?

The new deployment is using the current main branch of the beiwe-backend repo and Python 3.8 running on 64bit Amazon Linux 2. Of note, I've tried both with and without the code referenced in step 2 of the Configuring SSL section of the instructions (https://github.com/onnela-lab/beiwe-backend/wiki/Deployment-Instructions---Scalable-Deployment#configuring-ssl), and seem to get the same results both ways. 

I wasn't sure if this might be related to https://github.com/onnela-lab/beiwe-backend/issues/156 or some other open issue.

Anybody have any ideas about what might be happening here?

**Comment by biblicabeebli** [comment]
#156 shouldn't be related (I really need to get to that and clean issues on this repo generally).

The proxy error it is referencing is the apache -> uwsgi server-local proxy.  Apache accepts the incoming network connection, uwsgi runs the Beiwe code and handles threading.  The python 3.8 AWS platform upgrade was complex because the 3.6 platform used Apache ModWSGI, it didn't have a proxy at all.

Can you confirm you have done an `eb deploy` operation?

**Comment by biblicabeebli** [comment]
(your requests are passing through the load balancer and hitting the server - that's confirmation 1 that this is the local proxy - the curl is a second confirmation.)

**Comment by er42** [comment]
Thanks for the response!

Yes, this is happening after an `eb deploy`.

**Comment by biblicabeebli** [comment]
okay, couple things to test:

- check if uswgi is running?  I like htop, it should be installed. If you run it, set it to tree view by hitting `t` and then go up and down with the arrow keys you should be able to find it, or `f4` will let you search process names.
- there's a `logs` bash alias that `tail -f`'s all the relevant log files. .run that command in a shell session and see if anything useful comes up when you try these next two things.  (There are several log files, but it should be clear which ones are being reported.)  It's hard to say which one would be most relevant, and the AWS/EB-specific log files are verbose. No way around that.  
- watch the logs while running an eb deploy operation.
- watch the logs when you go to the online console, navigate to your EB application, and click on the actions button in the upper right of the page, and then run the "restart application server(s)" option - this does not restart the server, it runs a command on the server to restart apache (and uswgi, I think).

502 means that a network error occurred (server-local network stack, still counts as network).  That means something like uwsgi is not running, uwsgi is _segfaulting_ (e.g. its not our python that is failing), or one of apache or uwsgi is somehow misconfigured at the network level like by looking at the wrong ports or something.  The uswgi configuration is contained in the uwsg-config.yml file in the root of the repo, our Procfile simply tells uwsgi to run using the config file.

The commands that run as part of `eb deploy` are contained in the `.ebextensions/01.config` file, and the logs will tell us if one of those commands is failing.  If you don't have the `logs` alias then something is going wrong with the `eb deploy` and it isn't hitting any of the "container commands".  (generic commands and yum installs occur first, then it moves the uploaded code into place and runs "container commands" in the context of the repo and its python environment.)

If you want to deploy modified code you can do that by making changes, _staging_ that change with a `git add the-file` command, and then running an `eb deploy --staged`.

Hopefully that it detailed enough to get you either running or to an obvious step that is failing.  If you are able to get this working please report back on what happened.

(Oh, and there's an update to the server, customizable accelerometer and gyro sampling frequencies, that we are testing right now in beta builds of the app.  The backend component will be merged into main next week, so you will want to update the server when that happens.)

**Comment by er42** [comment]
Thanks! I'll try those troubleshooting steps soon and report back...

**Comment by er42** [comment]
So, this was very helpful in narrowing down the problem...

I do not get any errors/informative output on `eb deploy` nor any on restarting the application within the EB console. _However_,  uswgi is not running, and the logs alias does not exist on my EC2 instance. 

So, it sounds like this points to `eb deploy` not getting through everything in `.ebextensions/01.config`? 

I have no idea why that would be the case; I'm using the exact version of the file on the main repo, and did the deployment in the standard way. Any ideas? The only thing I noticed in looking through the AWS docs (https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/customize-containers-ec2.html) is that EB appears to want the container commands in quotes, but the ones in the config file are unquoted. Could that be the problem? (not sure if this is a recent change to the config file formatting). Any other ideas about what might be going on?

We'll be happy to update to the updated code to support customizable sampling frequencies once that's merged in!

**Comment by biblicabeebli** [comment]
It's not a formatting issue with the file, it is functional on all other deployments.

The first command that runs is the creation of a swapfile at `/swapfile`, if that file is not present on your server then your actions using the `eb` tool are not hitting the server at all.  That indicates configuration error or general credentialing issue with `eb`, or maybe you are pointing at the wrong elastic beanstalk instance, or aws region etc.  Until you get logs with meaningful content in them from basic operations like a deploy or clicking that restart button there isn't much I can do.

To help you out tho, here is the log alias; you can past this into ssh session and then just type `logs` to run it.
```
alias log='sudo tail -f /var/log/httpd/access_log /var/log/httpd/error_log /var/log/cfn-* /var/log/eb-* /var/log/messages'
```
In order, those files are:
- apache access log
- apache error log
- (multiple log files starting with cfn-) these are the AWS cloudwatch etc. logs
- (multiple log files starting with eb-) these are the Elastic Beanstalk operation logs
- the uwsgi logs and any python print statements from normal code operations.


I guess you might as well assemble some extra details like the elastic beanstalk version and environment type that was deployed.  Maybe something weird changed in the values the AWS APIs return from inside the launch script, and you've got like a java server deployed instead a python server.  (this is 100% a hail mary, I have no other ideas without observing directly.)

**Comment by er42** [comment]
Hmm, ok. Thanks for these suggestions.  I've been doing some more debugging but it's still not clear what exactly is going on. 

First of all, there is a 4GB swapfile at the location you described, so that seems like a good sign. 

I've checked the logs using the bash one-liner that you sent. Across all the log files, there are only a handful of things that stand out:

1. There are lots of proxy error messages in the httpd logs, as noted in my first post; no change there
2. There's some weird stuff going on in`/var/log/messages` . That log records a bunch of exceptions being raised about the Sentry DSNs being invalid, among other things (see below for a snippet). Per the instructions, we didn't attempt to include Sentry DSNs since it sounds like the new-format ones are not yet supported. Thus, we just kept the placeholder values in the `environment_configuration` file. So, obviously they're invalid, but that shouldn't be the thing causing this issue, right?
3. The `cfn-init` logs seem to suggest that the EB deployment went ok (see below)
4. The main thing that is conspicuously absent are any logs related to uwsgi, since that seems to have failed to ever start for reasons that remain unclear.
5. All the other logs look ok to me... lots of Code 200s being returned in the `cfn-wire` logs, etc.

Does that provide any useful clues? It seems like everything is pointing toward something going wrong with uwsgi, but I don't know anything about how that's supposed to work when it isn't broken, so unfortunately I don't have a clear idea of how best to troubleshoot it. Is there a way to start it manually by SSHing into the instance, for example, to see if there is some kind of helpful error message when it tries to start up that isn't showing up in the logs for some reason?

This is happening on EB version `app-6b1d-221105_140957096213`, with the platform `Python 3.8 running on 64bit Amazon Linux 2`. 

Here are some excerpts from the logs that look possibly important. Let me know if you want me to DM you the complete output of the logs command...

```
==> /var/log/messages <==
Nov  9 01:20:01 ip-XXX-XX-X-XXX systemd: Created slice User Slice of root.
Nov  9 01:20:01 ip-XXX-XX-X-XXX systemd: Started Session 1278 of user root.
Nov  9 01:20:01 ip-XXX-XX-X-XXX systemd: Removed slice User Slice of root.
Nov  9 01:20:06 ip-XXX-XX-X-XXX dhclient[2030]: XMT: Solicit on eth0, interval 108280ms.
Nov  9 01:21:54 ip-XXX-XX-X-XXX dhclient[2030]: XMT: Solicit on eth0, interval 126300ms.
Nov  9 01:24:00 ip-XXX-XX-X-XXX dhclient[2030]: XMT: Solicit on eth0, interval 131460ms.
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: Traceback (most recent call last):
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/django/core/handlers/wsgi.py", line 131, in __call__
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: signals.request_started.send(sender=self.__class__, environ=environ)
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/django/dispatch/dispatcher.py", line 180, in send
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: return [
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/django/dispatch/dispatcher.py", line 181, in <listcomp>
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: (receiver, receiver(signal=self, sender=sender, **named))
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/contrib/django/models.py", line 214, in before_request
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: self.client.context.activate()
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/contrib/django/models.py", line 54, in <lambda>
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: __getattr__ = lambda x, o: getattr(get_client(), o)
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/contrib/django/models.py", line 134, in get_client
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: instance = Client(**options)
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/contrib/django/client.py", line 151, in __init__
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: Client.__init__(self, *args, **kwargs)
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/base.py", line 171, in __init__
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: self.set_dsn(dsn, transport)
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/base.py", line 262, in set_dsn
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: result = RemoteConfig.from_string(
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/conf/remote.py", line 115, in from_string
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: raise InvalidDsn('Invalid Sentry DSN: %r' % url.geturl())
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: raven.exceptions.InvalidDsn: Invalid Sentry DSN: 'https://XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX@sentry.io/######'
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: Traceback (most recent call last):
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/django/core/handlers/wsgi.py", line 131, in __call__
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: signals.request_started.send(sender=self.__class__, environ=environ)
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/django/dispatch/dispatcher.py", line 180, in send
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: return [
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/django/dispatch/dispatcher.py", line 181, in <listcomp>
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: (receiver, receiver(signal=self, sender=sender, **named))
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/contrib/django/models.py", line 214, in before_request
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: self.client.context.activate()
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/contrib/django/models.py", line 54, in <lambda>
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: __getattr__ = lambda x, o: getattr(get_client(), o)
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/contrib/django/models.py", line 134, in get_client
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: instance = Client(**options)
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/contrib/django/client.py", line 151, in __init__
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: Client.__init__(self, *args, **kwargs)
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/base.py", line 171, in __init__
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: self.set_dsn(dsn, transport)
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/base.py", line 262, in set_dsn
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: result = RemoteConfig.from_string(
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/conf/remote.py", line 115, in from_string
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: raise InvalidDsn('Invalid Sentry DSN: %r' % url.geturl())
Nov  9 01:25:13 ip-XXX-XX-X-XXX web: raven.exceptions.InvalidDsn: Invalid Sentry DSN: 'https://XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX@sentry.io/######'
Nov  9 01:25:55 ip-XXX-XX-X-XXX web: Traceback (most recent call last):
Nov  9 01:25:55 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/django/core/handlers/wsgi.py", line 131, in __call__
Nov  9 01:25:55 ip-XXX-XX-X-XXX web: signals.request_started.send(sender=self.__class__, environ=environ)
Nov  9 01:25:55 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/django/dispatch/dispatcher.py", line 180, in send
Nov  9 01:25:55 ip-XXX-XX-X-XXX web: return [
Nov  9 01:25:55 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/django/dispatch/dispatcher.py", line 181, in <listcomp>
Nov  9 01:25:55 ip-XXX-XX-X-XXX web: (receiver, receiver(signal=self, sender=sender, **named))
Nov  9 01:25:55 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/contrib/django/models.py", line 214, in before_request
Nov  9 01:25:55 ip-XXX-XX-X-XXX web: self.client.context.activate()
Nov  9 01:25:55 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/contrib/django/models.py", line 54, in <lambda>
Nov  9 01:25:55 ip-XXX-XX-X-XXX web: __getattr__ = lambda x, o: getattr(get_client(), o)
Nov  9 01:25:55 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/contrib/django/models.py", line 134, in get_client
Nov  9 01:25:55 ip-XXX-XX-X-XXX web: instance = Client(**options)
Nov  9 01:25:55 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/contrib/django/client.py", line 151, in __init__
Nov  9 01:25:55 ip-XXX-XX-X-XXX web: Client.__init__(self, *args, **kwargs)
Nov  9 01:25:55 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/base.py", line 171, in __init__
Nov  9 01:25:55 ip-XXX-XX-X-XXX web: self.set_dsn(dsn, transport)
Nov  9 01:25:55 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/base.py", line 262, in set_dsn
Nov  9 01:25:55 ip-XXX-XX-X-XXX web: result = RemoteConfig.from_string(
Nov  9 01:25:55 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/conf/remote.py", line 115, in from_string
Nov  9 01:25:55 ip-XXX-XX-X-XXX web: raise InvalidDsn('Invalid Sentry DSN: %r' % url.geturl())
Nov  9 01:25:55 ip-XXX-XX-X-XXX web: raven.exceptions.InvalidDsn: Invalid Sentry DSN: 'https://XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX@sentry.io/######'
Nov  9 01:25:56 ip-XXX-XX-X-XXX web: Traceback (most recent call last):
Nov  9 01:25:56 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/django/core/handlers/wsgi.py", line 131, in __call__
Nov  9 01:25:56 ip-XXX-XX-X-XXX web: signals.request_started.send(sender=self.__class__, environ=environ)
Nov  9 01:25:56 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/django/dispatch/dispatcher.py", line 180, in send
Nov  9 01:25:56 ip-XXX-XX-X-XXX web: return [
Nov  9 01:25:56 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/django/dispatch/dispatcher.py", line 181, in <listcomp>
Nov  9 01:25:56 ip-XXX-XX-X-XXX web: (receiver, receiver(signal=self, sender=sender, **named))
Nov  9 01:25:56 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/contrib/django/models.py", line 214, in before_request
Nov  9 01:25:56 ip-XXX-XX-X-XXX web: self.client.context.activate()
Nov  9 01:25:56 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/contrib/django/models.py", line 54, in <lambda>
Nov  9 01:25:56 ip-XXX-XX-X-XXX web: __getattr__ = lambda x, o: getattr(get_client(), o)
Nov  9 01:25:56 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/contrib/django/models.py", line 134, in get_client
Nov  9 01:25:56 ip-XXX-XX-X-XXX web: instance = Client(**options)
Nov  9 01:25:56 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/contrib/django/client.py", line 151, in __init__
Nov  9 01:25:56 ip-XXX-XX-X-XXX web: Client.__init__(self, *args, **kwargs)
Nov  9 01:25:56 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/base.py", line 171, in __init__
Nov  9 01:25:56 ip-XXX-XX-X-XXX web: self.set_dsn(dsn, transport)
Nov  9 01:25:56 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/base.py", line 262, in set_dsn
Nov  9 01:25:56 ip-XXX-XX-X-XXX web: result = RemoteConfig.from_string(
Nov  9 01:25:56 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/conf/remote.py", line 115, in from_string
Nov  9 01:25:56 ip-XXX-XX-X-XXX web: raise InvalidDsn('Invalid Sentry DSN: %r' % url.geturl())
Nov  9 01:25:56 ip-XXX-XX-X-XXX web: raven.exceptions.InvalidDsn: Invalid Sentry DSN: 'https://XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX@sentry.io/######'
Nov  9 01:26:00 ip-XXX-XX-X-XXX web: Traceback (most recent call last):
Nov  9 01:26:00 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/django/core/handlers/wsgi.py", line 131, in __call__
Nov  9 01:26:00 ip-XXX-XX-X-XXX web: signals.request_started.send(sender=self.__class__, environ=environ)
Nov  9 01:26:00 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/django/dispatch/dispatcher.py", line 180, in send
Nov  9 01:26:00 ip-XXX-XX-X-XXX web: return [
Nov  9 01:26:00 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/django/dispatch/dispatcher.py", line 181, in <listcomp>
Nov  9 01:26:00 ip-XXX-XX-X-XXX web: (receiver, receiver(signal=self, sender=sender, **named))
Nov  9 01:26:00 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/contrib/django/models.py", line 214, in before_request
Nov  9 01:26:00 ip-XXX-XX-X-XXX web: self.client.context.activate()
Nov  9 01:26:00 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/contrib/django/models.py", line 54, in <lambda>
Nov  9 01:26:00 ip-XXX-XX-X-XXX web: __getattr__ = lambda x, o: getattr(get_client(), o)
Nov  9 01:26:00 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/contrib/django/models.py", line 134, in get_client
Nov  9 01:26:00 ip-XXX-XX-X-XXX web: instance = Client(**options)
Nov  9 01:26:00 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/contrib/django/client.py", line 151, in __init__
Nov  9 01:26:00 ip-XXX-XX-X-XXX web: Client.__init__(self, *args, **kwargs)
Nov  9 01:26:00 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/base.py", line 171, in __init__
Nov  9 01:26:00 ip-XXX-XX-X-XXX web: self.set_dsn(dsn, transport)
Nov  9 01:26:00 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/base.py", line 262, in set_dsn
Nov  9 01:26:00 ip-XXX-XX-X-XXX web: result = RemoteConfig.from_string(
Nov  9 01:26:00 ip-XXX-XX-X-XXX web: File "/var/app/venv/staging-LQM1lest/lib/python3.8/site-packages/raven/conf/remote.py", line 115, in from_string
Nov  9 01:26:00 ip-XXX-XX-X-XXX web: raise InvalidDsn('Invalid Sentry DSN: %r' % url.geturl())
Nov  9 01:26:00 ip-XXX-XX-X-XXX web: raven.exceptions.InvalidDsn: Invalid Sentry DSN: 'https://XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX@sentry.io/######'
Nov  9 01:26:12 ip-XXX-XX-X-XXX dhclient[2030]: XMT: Solicit on eth0, interval 128710ms.
Nov  9 01:28:21 ip-XXX-XX-X-XXX dhclient[2030]: XMT: Solicit on eth0, interval 114340ms.
Nov  9 01:30:01 ip-XXX-XX-X-XXX systemd: Created slice User Slice of root.
Nov  9 01:30:01 ip-XXX-XX-X-XXX systemd: Started Session 1280 of user root.
Nov  9 01:30:01 ip-XXX-XX-X-XXX systemd: Started Session 1279 of user root.
Nov  9 01:30:01 ip-XXX-XX-X-XXX systemd: Removed slice User Slice of root.
Nov  9 01:30:15 ip-XXX-XX-X-XXX dhclient[2030]: XMT: Solicit on eth0, interval 108380ms.


==> /var/log/cfn-init.log <==
2022-11-05 21:10:27,104 [INFO] Running configSets: Infra-EmbeddedPostBuild
2022-11-05 21:10:27,106 [INFO] Running configSet Infra-EmbeddedPostBuild
2022-11-05 21:10:27,108 [INFO] Running config postbuild_0_beiwe_application
2022-11-05 21:10:27,122 [INFO] Command 01_setup_profile succeeded
2022-11-05 21:10:27,134 [INFO] Command 02_setup_reasonable_inputrc succeeded
2022-11-05 21:10:43,306 [INFO] Command 03_update_pglib succeeded
2022-11-05 21:10:44,753 [INFO] Command 04_migrate succeeded
2022-11-05 21:10:45,602 [INFO] Command 05_run_optional_deploy_script succeeded
2022-11-05 21:10:45,603 [INFO] ConfigSets completed
2022-11-05 21:10:45,603 [INFO] -----------------------Build complete-----------------------

```

**Comment by biblicabeebli** [comment]
> There's some weird stuff going on in/var/log/messages . That log records a bunch of exceptions being raised about the Sentry DSNs being invalid, among other things (see below for a snippet). Per the instructions, we didn't attempt to include Sentry DSNs since it sounds like the new-format ones are not yet supported. Thus, we just kept the placeholder values in the environment_configuration file. So, obviously they're invalid, but that shouldn't be the thing causing this issue, right?

It's this.

Unless sentry has changed their format _yet again for a 4th time_ new style dsns are absolutely supported.  We do need to update the underlying library to their newer one, but the DSNs work.

This is a cleaned up view of the stack trace. The error occurs at django load-time, our codebase never even loads, and since it is an error in the error reporting framework configuration it can't reported.  (Talk about worst possible failure mode...)
```
File "site-packages/django/core/handlers/wsgi.py", line 131, in __call__
    signals.request_started.send(sender=self.__class__, environ=environ)
File "site-packages/django/dispatch/dispatcher.py", line 180, in send
    return [
File "site-packages/django/dispatch/dispatcher.py", line 181, in <listcomp>
    (receiver, receiver(signal=self, sender=sender, **named))
File "site-packages/raven/contrib/django/models.py", line 214, in before_request
    self.client.context.activate()
File "site-packages/raven/contrib/django/models.py", line 54, in <lambda>
    __getattr__ = lambda x, o: getattr(get_client(), o)
File "site-packages/raven/contrib/django/models.py", line 134, in get_client
    instance = Client(**options)
File "site-packages/raven/contrib/django/client.py", line 151, in __init__
    Client.__init__(self, *args, **kwargs)
File "site-packages/raven/base.py", line 171, in __init__
    self.set_dsn(dsn, transport)
File "site-packages/raven/base.py", line 262, in set_dsn
    result = RemoteConfig.from_string(
File "site-packages/raven/conf/remote.py", line 115, in from_string
    raise InvalidDsn('Invalid Sentry DSN: %r' % url.geturl())
raven.exceptions.InvalidDsn: Invalid Sentry DSN: 'https://XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX@sentry.io/######'
```

You are using the placeholder DSN.  You need to provide a real DSN from sentriy.io, or you can remove the `SENTRY_ELASTIC_BEANSTALK_DSN` key (that one specifically but the others too) from the AWS console server configuration to disable sentry entirely.

(If you sanitized that DSN value before pasting it please let me know, and also provide a sample so that I can build some fault handling into the settings code.)

**Comment by er42** [comment]
Aha -- I plugged in the current-style DSN from Sentry, and now it works! So you're right; that was indeed the problem.

Might I suggest an update to the wiki deployment instructions? Currently the text reads:

_Beiwe is currently using a very old format for Sentry DSNs. Until we have updated Beiwe any DSNs you provide may not work. However, the deployment script should tell you that your new-style DSN does not match the old DSN format, **so for now just run without those credentials. We will be updating the Sentry support to use the new DSN format**._ 
(emphasis mine)

It was because of this that I thought it was unnecessary/useless to replace the dummy DSN values in the environment variables file, though I now understand it's critical. Clarifying these few sentences might save others who self-deploy the code from making the same error that I did.

Anyway, I'm going to go ahead and close the issue since it's resolved now. Thanks so much for your help in figuring this out!
